config/clusters/maap/staging.values.yaml

```diff
@@ -280,6 +280,24 @@ jupyterhub:
         - CPU:XXXL
         - GPU:T4
     extraConfig:
+      001-username-claim: |
+        from oauthenticator.generic import GenericOAuthenticator
+
+        def populate_token(spawner, auth_state):
+          # For our deployment-service-check health check user, there is no auth_state.
+          # So these env variables need not be set.
+          if auth_state:
+
+            spawner.environment.update({
+              "MAAP_PGT": f"jwt:{auth_state.get("id_token", "")}",
+              "KC_ACCESS_TOKEN": auth_state.get("access_token", ""),
+              "KC_ID_TOKEN": auth_state.get("id_token", ""),
+              "KC_REFRESH_TOKEN": auth_state.get("refresh_token", "")
+            })
+
+        c.Spawner.auth_state_hook = populate_token
+
+        c.JupyterHub.authenticator_class = GenericOAuthenticator
```

Comment:
This is also not required

Reply:
Just the two lines I commented on need to be removed. The rest shall stay. Auth0 uses a custom authenticator, you don’t need just that bit